### DISCUSSION TITLE
Lesson 7: Question about Interactions.s.sol

### DISCUSSION THREAD

**Discussion by Ifra001** [Cyfrin/foundry-full-course-cu#1847] [Q&A] [answered]
Patrick made an ```Interactions.s.sol``` in Lesson 7 but I don't get why? Is there some additional easiness? Also, I forgot what the ```vm.startbroadcast``` and ```vm.stopbroadcast``` did.😅

**Comment by urosognjenovic** [comment, marked as answer]
He created ```Interactions.s.sol``` to programatically interact with the contracts instead of manually funding and withdrawing.

Everything between ```vm.startBroadcast()``` and ```vm.stopBroadcast()``` is sent onchain. If you navigate to the ```Vm.sol``` contract you'll find this comment above the ```vm.startBroadcast()``` function: "Has all subsequent calls (at this call depth only) create transactions that can later be signed and sent onchain."

**Reply by Ifra001** [reply]
Thank you bro, things like these slip out of my mind sometimes so I am glad that you helped out!

**Reply by urosognjenovic** [reply]
No problem! Here to help!
